src/main/java/com/autotune/analyzer/recommendations/updater/vpa/VpaUpdaterImpl.java

```diff
@@ -0,0 +1,377 @@
+                            AnalyzerConstants.RecommendationItem.MEMORY, containerName, memoryRecommendationValue));
+
+                    String cpuRecommendationValueForVpa = RecommendationUtils.resource2str(AnalyzerConstants.RecommendationItem.CPU.toString(),
+                            cpuRecommendationValue);
+                    String memoryRecommendationValueForVpa = RecommendationUtils.resource2str(AnalyzerConstants.RecommendationItem.MEMORY.toString(),
+                            memoryRecommendationValue);
+
+                    // creating container resource vpa object
+                    RecommendedContainerResources recommendedContainerResources = new RecommendedContainerResources();
+                    recommendedContainerResources.setContainerName(containerName);
+
+                    // setting target values
+                    Map<String, Quantity> target = new HashMap<>();
+                    target.put(AnalyzerConstants.RecommendationItem.CPU.toString(), new Quantity(cpuRecommendationValueForVpa));
+                    target.put(AnalyzerConstants.RecommendationItem.MEMORY.toString(), new Quantity(memoryRecommendationValueForVpa));
+
+                    // setting lower bound values
+                    Map<String, Quantity> lowerBound = new HashMap<>();
+                    lowerBound.put(AnalyzerConstants.RecommendationItem.CPU.toString(), new Quantity(cpuRecommendationValueForVpa));
+                    lowerBound.put(AnalyzerConstants.RecommendationItem.MEMORY.toString(), new Quantity(memoryRecommendationValueForVpa));
+
+                    // setting upper bound values
+                    Map<String, Quantity> upperBound = new HashMap<>();
+                    upperBound.put(AnalyzerConstants.RecommendationItem.CPU.toString(), new Quantity(cpuRecommendationValueForVpa));
+                    upperBound.put(AnalyzerConstants.RecommendationItem.MEMORY.toString(), new Quantity(memoryRecommendationValueForVpa));
+
+                    recommendedContainerResources.setLowerBound(lowerBound);
+                    recommendedContainerResources.setTarget(target);
+                    recommendedContainerResources.setUpperBound(upperBound);
+
+                    containerRecommendations.add(recommendedContainerResources);
+                }
+            }
+        }
+        return containerRecommendations;
+    }
+
+    /*
+     * Creates a Vertical Pod Autoscaler (VPA) object in the specified namespace
+     * for the given deployment and containers.
+     */
+    public void createVpaObject(KruizeObject kruizeObject) throws UnableToCreateVPAException {
+        try {
+            // checks if updater is installed or not
+            if (isUpdaterInstalled()) {
+                LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.CREATEING_VPA, kruizeObject.getExperimentName()));
```

Comment:
This might create too much logging if we are apply VPA for every object in the cluster. I'd suggest make it LOGGER.debug. Only error's should be logged by default

Reply:
Agree. I've updated the `info` messages to `debug` messages which are experiment specific. Thanks for pointing out.